paddle/framework/operator.h

```diff
@@ -83,8 +83,7 @@ class OperatorBase {
   virtual std::string DebugString() const;
 
   /// Net will call this function to Run an op.
-  virtual void Run(const Scope& scope,
-                   const platform::DeviceContext& dev_ctx) const = 0;
+  virtual void Run(const Scope& scope, const platform::Place& place) const = 0;
```

Comment:
Currently, we can simply add another interface here:

```
virtual void Run(const Scope& scope,
                           const platform::DeviceContext& dev_ctx) const = 0;

void Run(const Scope& scope, const platform::Place& place) const {
    platform::DeviceContextPool &pool = platform::DeviceContextPool::Get();
    auto &dev_ctx = *pool.Borrow(place);
    Run(scope, dev_ctx);
}
```

Reply:
We hold the same believe that Operatorbase has one and only one runnable interface. 
Now it's the early time that we can fix operator interface once, otherwise, we can not simply remove it.

Reply:
The same confusing with @QiJune , for this PR, too many operators will fetch the `device_ctx` and add the following code:
```cpp
// get device context from pool
platform::DeviceContextPool &pool = platform::DeviceContextPool::Get();
auto &dev_ctx = *pool.Borrow(place);
```

I knew from @dzhwinter  that one Op without Kernel would fetch the `dev_ctx`  from `DeviceContextPool`,  and I think the suggestion from @QiJune maybe a simple way, and don't add too much repeated code in the Ops.

Reply:
I hold a different view. @QiJune 's suggestion is a transitional approach. 

DeviceContextPool contains device related resources, so I believe only the OperatorWithKernel need to touch the pool.

```c++
void OperatorWithKernel::Run(const Scope& scope,
                             const platform::Place& place) const {
  RuntimeInferShapeContext infer_shape_ctx(*this, scope);
  this->InferShape(&infer_shape_ctx);
  platform::DeviceContextPool& pool = platform::DeviceContextPool::Get();
  auto dev_ctx = pool.Borrow(place);
```
If we make an agreement on above, so we should not add a Run(DeviceContext) in the final solution.
In addition, most of the operators don't need to add the redundant snippet of get device from the global pool, so why this PR contains a lot `Get()` ? 
The reason is our CopyFrom didn't different CPUDevice and GPUDevice, actually, only the GPUDevice need to get a devicecontext from the pool and do a copy. We will change it in the future.